### PR TITLE
[improve][UT] Upgrade junit to 5.+

### PR DESCRIPTION
**Pull request by ashulin**
## Purpose of this pull request
Right now, we still use junit 4.+ in our repo, I hope we can upgrade to [Junit 5.+](https://junit.org/junit5/docs/current/user-guide/#writing-tests), then we can easy to write the UT, JUnit 5 provide some useful feature, e.g. @ParameterizedTest, @EnabledOnOs,...
https://github.com/apache/incubator-seatunnel/issues/2283

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by ashulin** [comment]
Please trigger the Github action again. @CalvinKirs @Hisoka-X

File: tools/dependencies/known-dependencies.txt [hunk 1]
```diff
@@ -28,7 +28,7 @@ apacheds-i18n-2.0.0-M15.jar
 apacheds-kerberos-codec-2.0.0-M15.jar
 api-asn1-api-1.0.0-M20.jar
 api-util-1.0.0-M20.jar
-apiguardian-api-1.1.0.jar
+apiguardian-api-1.1.2.jar
```
Comment (CalvinKirs) [inline tools/dependencies/known-dependencies.txt:31]: hi, Ilooked this PR and it's good overall, but I have a question, the jars we introduce are all at the test level, why is there a jar upgrade? Can you check where it was imported?

Comment (ashulin) [reply tools/dependencies/known-dependencies.txt:31]: It upgrades because of `junit-jupiter-engine` in Junit 5

Comment (CalvinKirs) [reply tools/dependencies/known-dependencies.txt:31]: got it. It seems that other components also depend on this jar. Please follow this document to add the corresponding license changes.

Comment (CalvinKirs) [reply tools/dependencies/known-dependencies.txt:31]: https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**